pkg/apiserver/profiling/pprof.go

```diff
@@ -3,170 +3,60 @@
+	tmpPath, profileOutputType, err := fetcher.FetchAndWriteToFile(op.duration, op.fileNameWithoutExt)
 	if err != nil {
-		return "", fmt.Errorf("failed to get DOT output from file: %v", err)
+		return "", "", fmt.Errorf("failed to fetch annd write to temp file: %v", err)
 	}
 
-	b, err := ioutil.ReadFile(filepath.Clean(f))
-	if err != nil {
-		return "", fmt.Errorf("failed to get DOT output from file: %v", err)
-	}
-
-	tmpfile, err := ioutil.TempFile("", op.fileNameWithoutExt)
-	if err != nil {
-		return "", fmt.Errorf("failed to create temp file: %v", err)
-	}
-	defer tmpfile.Close() // #nosec
-	tmpPath := fmt.Sprintf("%s.%s", tmpfile.Name(), "svg")
-
-	g := graphviz.New()
-	mu.Lock()
-	defer mu.Unlock()
-	graph, err := graphviz.ParseBytes(b)
-	if err != nil {
-		return "", fmt.Errorf("failed to parse DOT file: %v", err)
-	}
-
-	if err := g.RenderFilename(graph, graphviz.SVG, tmpPath); err != nil {
-		return "", fmt.Errorf("failed to render SVG: %v", err)
-	}
-
-	return tmpPath, nil
+	return tmpPath, profileOutputType, nil
 }
 
-type flagSet struct {
-	*flag.FlagSet
-	args []string
+type fetcher struct {
+	target         *model.RequestTargetNode
+	profileFetcher *profileFetcher
 }
 
-func fetchPprof(op *pprofOptions, format string) (string, error) {
-	tmpfile, err := ioutil.TempFile("", op.fileNameWithoutExt)
+func (f *fetcher) FetchAndWriteToFile(duration uint, fileNameWithoutExt string) (string, TaskProfileOutputType, error) {
+	tmpfile, err := ioutil.TempFile("", fileNameWithoutExt)
 	if err != nil {
-		return "", fmt.Errorf("failed to create temp file: %v", err)
-	}
-	defer tmpfile.Close() // #nosec
-	tmpPath := fmt.Sprintf("%s.%s", tmpfile.Name(), format)
-	format = "-" + format
-	args := []string{
-		format,
-		// prevent printing stdout
-		"-output", "dummy",
-		"-seconds", strconv.Itoa(int(op.duration)),
-	}
-	address := fmt.Sprintf("%s:%d", op.target.IP, op.target.Port)
-	args = append(args, address)
-	f := &flagSet{
-		FlagSet: flag.NewFlagSet("pprof", flag.PanicOnError),
-		args:    args,
+		return "", "", fmt.Errorf("failed to create tmpPath to write profile: %v", err)
 	}
-	if err := driver.PProf(&driver.Options{
-		Fetch:   &fetcher{profileFetcher: op.fetcher, target: op.target},
-		Flagset: f,
-		UI:      &blankPprofUI{},
-		Writer:  &oswriter{output: tmpPath},
-	}); err != nil {
-		return "", fmt.Errorf("failed to generate profile report: %v", err)
-	}
-	return tmpPath, nil
-}
-
-func (f *flagSet) StringList(o, d, c string) *[]*string {
-	return &[]*string{f.String(o, d, c)}
-}
-
-func (f *flagSet) ExtraUsage() string {
-	return ""
-}
 
-func (f *flagSet) Parse(usage func()) []string {
-	f.Usage = usage
-	_ = f.FlagSet.Parse(f.args)
-	return f.Args()
-}
-
-func (f *flagSet) AddExtraUsage(eu string) {}
-
-// oswriter implements the Writer interface using a regular file.
-type oswriter struct {
-	output string
-}
+	defer func() {
+		if err := tmpfile.Close(); err != nil {
+			fmt.Printf("failed to close file, %v", err)
```

Comment:
You can use `log.Warn`, since `fmt.Printf` will break the output log format.

Or you can just discard this error :) In the rest of the code base this kind of error is usually discarded.